packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,171 @@
+  expect(addCaregiverDialog).toBeVisible()
+  expect(within(addCaregiverDialog).getByText('Add a caregiver')).toBeVisible()
+  const emailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+  expect(emailInput).toBeVisible()
+
+  const newCaregiverEmail = 'new-caregiver@mail.com'
+  await userEvent.type(emailInput, 'new-caregiver@mail.com')
+  const inviteButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+  expect(inviteButton).toBeEnabled()
+
+  jest.spyOn(NotificationApi, 'getSentInvitations').mockResolvedValueOnce([{
+    email: newCaregiverEmail,
+    type: NotificationType.directInvitation,
+    target: { id: 'target-id' },
+    id: 'my-id'
+  } as Notification])
+  await userEvent.click(inviteButton)
+
+  expect(screen.queryByTestId('patient-add-caregiver-dialog')).not.toBeInTheDocument()
+  const inviteSuccessfulSnackbar = screen.getByTestId('alert-snackbar')
+  expect(inviteSuccessfulSnackbar).toHaveTextContent('Invite sent!')
+  await userEvent.click(within(inviteSuccessfulSnackbar).getByTitle('Close'))
+
+  const caregiversTableAfterAdd = await screen.findByTestId('patient-caregivers-list')
+  expect(caregiversTableAfterAdd).toBeVisible()
+
+  const caregiverRowAfterAdd = within(caregiversTableAfterAdd).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRowAfterAdd).toBeVisible()
+  expect(caregiverRowAfterAdd).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRowAfterAdd).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
+
+  const newCaregiverRow = within(caregiversTableAfterAdd).getByTestId(`patient-caregivers-table-row-${newCaregiverEmail}`)
+  expect(newCaregiverRow).toBeVisible()
+  expect(newCaregiverRow).toHaveTextContent('--new-caregiver@mail.com')
+  const removeCaregiverButton = within(newCaregiverRow).getByTestId(`patient-caregivers-table-row-${newCaregiverEmail}-button-remove`)
+  expect(removeCaregiverButton).toBeVisible()
+
+  await userEvent.click(removeCaregiverButton)
+
+  const removeCaregiverDialog = screen.getByTestId('remove-direct-share-dialog')
+  expect(within(removeCaregiverDialog).getByText('Remove caregiver new-caregiver@mail.com')).toBeVisible()
+  expect(removeCaregiverDialog).toHaveTextContent('Are you sure you want to remove caregiver new-caregiver@mail.com?They will no longer have access to your data.')
+  const cancelButton = within(removeCaregiverDialog).getByRole('button', { name: 'Cancel' })
+  expect(cancelButton).toBeEnabled()
+  const removeButton = within(removeCaregiverDialog).getByRole('button', { name: 'Remove caregiver' })
+  expect(removeButton).toBeEnabled()
+
+  jest.spyOn(NotificationApi, 'getSentInvitations').mockResolvedValueOnce([])
+  jest.spyOn(DirectShareApi, 'getDirectShares').mockResolvedValue([{ status: UserInviteStatus.Accepted, user: mockCaregiverUser }])
+  await userEvent.click(removeButton)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏There should be a check to verify that the HTTP request was made with the correct parameters :slightly_smiling_face: